code/obj/item/mine.dm

```diff
@@ -220,23 +220,40 @@
 	icon_state = "mine_incendiary"
 
 	armed
-		armed = 1
+		armed = TRUE
 
 	custom_stuff(var/atom/M)
 		if (!src || !istype(src))
 			return
 
 		fireflash_sm(get_turf(src), 3, 3000, 500)
 		playsound(src.loc, 'sound/effects/bamf.ogg', 50, 1)
-		return
 
 /obj/item/mine/stun
 	name = "stun land mine"
 	desc = "An anti-personnel mine designed to stun its victim nonlethally."
 	icon_state = "mine_stun"
 
 	armed
-		armed = 1
+		armed = TRUE
+
+	nanotrasen
+		name = "NT stun land mine"
+		desc = "A Nanotrasen-brand land mine used to stun victims nonlethally."
+		icon_state = "mine_stun_nt"
+
+		custom_stuff(var/atom/M)
+			if (!src || !istype(src))
+				return
+
+			var/list/mobs = src.get_mobs_on_turf(1)
+			if (mobs.len)
+				for (var/mob/living/L in mobs)
+					if (istype(L))
+						L.changeStatus("weakened", 10 SECONDS)
+						L.stuttering += 10
```

Comment:
do_disorient instead of outright stun

Reply:
i'll fix this with regular mine code as well